cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java

```diff
@@ -396,11 +400,13 @@ public CloudVmTypes virtualMachinesForDistroX(CloudCredential cloudCredential, R
             for (Map.Entry<String, Set<VmType>> stringSetEntry : cloudVmResponses.entrySet()) {
                 returnVmResponses.put(stringSetEntry.getKey(), stringSetEntry.getValue().stream()
                         .filter(enabledDistroxInstanceTypeFilter)
+                        .filter(e -> minimalHardwareFilter.suitableAsMinimumHardware(e))
                         .collect(Collectors.toSet()));
             }
         } else {
             for (Map.Entry<String, Set<VmType>> stringSetEntry : cloudVmResponses.entrySet()) {
                 returnVmResponses.put(stringSetEntry.getKey(), stringSetEntry.getValue().stream()
+                        .filter(e -> minimalHardwareFilter.suitableAsMinimumHardware(e))
                         .collect(Collectors.toSet()));
             }
         }
```

Comment:
This comment is for lines 416-417 but GH does not let me add it there. So, `@NotNull` shall be removed (the agreement was that we were not going to use JetBrains annotations), and `getCloudVmTypes()` shall be made `private` (or package private if directly called by UT).